# 2020.03.28 (Sat)

## Base model: spaCy

In [1]:
#!prodigy train ner cord_19_abstracts_match models/en_core_web_lg_no_ner --output models/2020_03_28_match/en_rf_web_lg

✔ Loaded model 'models/en_core_web_lg_no_ner'
Created and merged data for 86 total examples
Using 43 train / 43 eval (split 50%)
Component: ner | Batch size: compounding | Dropout: 0.2 | Iterations: 10
ℹ Baseline accuracy: 0.000

=========================== ✨  Training the model ===========================

#    Loss       Precision   Recall     F-Score 
--   --------   ---------   --------   --------
 1    2260.23       3.822      5.714      4.580                                 
 2     304.20      96.203     72.381     82.609                                 
 3      35.94      93.069     89.524     91.262                                 
 4       7.71      91.262     89.524     90.385                                 
 5       6.89      94.000     89.524     91.707                                 
 6       6.26      94.949     89.524     92.157                                 
 7       4.15      94.000     89.524     91.707                                 
 8      10.58      92.157   

## Base model: scispaCy

In [2]:
#!prodigy train ner cord_19_abstracts_match models/en_core_sci_lg_no_ner --output models/2020_03_28_match/en_rf_sci_lg

✔ Loaded model 'models/en_core_sci_lg_no_ner'
Created and merged data for 86 total examples
Using 43 train / 43 eval (split 50%)
Component: ner | Batch size: compounding | Dropout: 0.2 | Iterations: 10
ℹ Baseline accuracy: 0.000

=========================== ✨  Training the model ===========================

#    Loss       Precision   Recall     F-Score 
--   --------   ---------   --------   --------
 1    1864.33       3.333      2.941      3.125                                 
 2     303.79      93.902     75.490     83.696                                 
 3      22.39      92.784     88.235     90.452                                 
 4      17.47      94.792     89.216     91.919                                 
 5       5.36      93.814     89.216     91.457                                 
 6       6.78      92.857     89.216     91.000                                 
 7       6.09      93.814     89.216     91.457                                 
 8       4.87      94.792   

# 2020.03.29 (Sun)

## Test the model trained on Sat

In [1]:
import spacy
import warnings
from pprint import pprint

def test_model(model_path):
    warnings.filterwarnings("ignore",category=UserWarning)
    
    nlp = spacy.load(model_path)
    texts = [
    "Known risk factors for the disease are: older age, male gender, diabetes and leukemia. Female patiens and children are less susceptible.",
    "Diabetes is a known risk factor.",
    "Leukemia is a risk factor, too."
    ]

    for text in texts:
        doc = nlp(text)

        spacy.displacy.render(doc, style='ent', jupyter=True)
        pprint([(ent.text, ent.label_) for ent in doc.ents])

In [2]:
test_model("models/2020_03_28_match/en_rf_web_lg")

[('age', 'RISK_FACTOR'), ('male', 'RISK_FACTOR'), ('diabetes', 'RISK_FACTOR')]


[('Diabetes', 'RISK_FACTOR')]


[]


Only the factors that appeared during the training are detected. _Leukemia_ is **not highlighed** and that means that the NER model did not learn to recognize risk factors based on the sentence structure.

## Train new models after the `teach` session

In [1]:
#!prodigy train ner cord_19_abstracts_teach models/en_core_web_lg_no_ner --output models/2020_03_29_teach/en_rf_web_lg

✔ Loaded model 'models/en_core_web_lg_no_ner'
Created and merged data for 393 total examples
Using 197 train / 196 eval (split 50%)
Component: ner | Batch size: compounding | Dropout: 0.2 | Iterations: 10
ℹ Baseline accuracy: 0.000

=========================== ✨  Training the model ===========================

#    Loss       Precision   Recall     F-Score 
--   --------   ---------   --------   --------
 1     333.36      89.669     92.735     91.176                                 
 2      64.66      92.982     90.598     91.775                                 
 3      53.47      94.492     95.299     94.894                                 
 4      45.26      94.492     95.299     94.894                                 
 5      29.42      94.958     96.581     95.763                                 
 6      30.44      96.170     96.581     96.375                                 
 7      23.82      95.745     96.154     95.949                                 
 8       5.78      95.319

In [2]:
#!prodigy train ner cord_19_abstracts_teach models/en_core_sci_lg_no_ner --output models/2020_03_29_teach/en_rf_sci_lg

✔ Loaded model 'models/en_core_sci_lg_no_ner'
Created and merged data for 393 total examples
Using 197 train / 196 eval (split 50%)
Component: ner | Batch size: compounding | Dropout: 0.2 | Iterations: 10
ℹ Baseline accuracy: 0.000

=========================== ✨  Training the model ===========================

#    Loss       Precision   Recall     F-Score 
--   --------   ---------   --------   --------
 1     289.28      90.991     91.818     91.403                                 
 2      54.12      92.237     91.818     92.027                                 
 3      84.39      91.593     94.091     92.825                                 
 4      46.45      93.213     93.636     93.424                                 
 5      19.43      95.000     95.000     95.000                                 
 6      15.25      93.665     94.091     93.878                                 
 7       4.07      93.363     95.909     94.619                                 
 8      11.23      93.722

In [3]:
test_model("models/2020_03_29_teach/en_rf_web_lg")

[('age', 'RISK_FACTOR'),
 ('male gender', 'RISK_FACTOR'),
 ('diabetes', 'RISK_FACTOR')]


[('Diabetes', 'RISK_FACTOR')]


[]


Better model scores, but _leukemia_ still **not detected**.

# 2020.03.30 (Mon)

## Starting the annotations from scratch (`strict`)

Terms like _male_ or _age_ will only be marked, when they are mentioned in the context of COVID-19 risk factors, i.e. not in articles about animals or children.

In [5]:
!prodigy ner.manual cord_19_abstracts_strict models/en_core_sci_lg_no_ner data/raw/cord_19_abstracts_filtered.jsonl \
  --label RISK_FACTOR --patterns patterns/RF_list_2020.03.17.20037572.jsonl

Using 1 label(s): RISK_FACTOR
Added dataset cord_19_abstracts_strict to database SQLite.

✨  Starting the web server at http://localhost:8080 ...
Open the app in your browser and start annotating!

^C

✔ Saved 41 annotations to database SQLite
Dataset: cord_19_abstracts_strict
Session ID: 2020-03-30_20-24-26



In [7]:
#!prodigy train ner cord_19_abstracts_strict models/en_core_sci_lg_no_ner --output models/2020_03_30_strict/en_rf_sci_lg

✔ Loaded model 'models/en_core_sci_lg_no_ner'
Created and merged data for 41 total examples
Using 21 train / 20 eval (split 50%)
Component: ner | Batch size: compounding | Dropout: 0.2 | Iterations: 10
ℹ Baseline accuracy: 0.000

=========================== ✨  Training the model ===========================

#    Loss       Precision   Recall     F-Score 
--   --------   ---------   --------   --------
 1    1737.49       0.000      0.000      0.000                                 
 2      25.99       0.000      0.000      0.000                                 
 3     115.63       0.000      0.000      0.000                                 
 4      24.86       0.000      0.000      0.000                                 
 5      22.36       0.000      0.000      0.000                                 
 6      18.33       0.000      0.000      0.000                                 
 7      20.52       0.000      0.000      0.000                                 
 8      28.45       0.000   

## Starting the annotations from scratch (`all RF`)

Highlight all risk factors - for any disease or species - but only is sentences that clear say it is a risk factor.

In [ ]:
!prodigy ner.manual cord_19_abstracts_all_rf models/en_core_sci_lg_no_ner data/raw/cord_19_abstracts_filtered.jsonl \
  --label RISK_FACTOR --patterns patterns/RF_highlight_factor_phrases.jsonl

In [6]:
!prodigy train ner cord_19_abstracts_all_rf models/en_core_sci_lg_no_ner --output models/2020_03_30_all_RF/en_rf_sci_lg

✔ Loaded model 'models/en_core_sci_lg_no_ner'
Created and merged data for 58 total examples
Using 29 train / 29 eval (split 50%)
Component: ner | Batch size: compounding | Dropout: 0.2 | Iterations: 10
ℹ Baseline accuracy: 0.000

=========================== ✨  Training the model ===========================

#    Loss       Precision   Recall     F-Score 
--   --------   ---------   --------   --------
 1    1960.54       0.000      0.000      0.000                                 
 2      95.38       0.000      0.000      0.000                                 
 3     161.78     100.000      2.128      4.167                                 
 4      88.78      40.000      4.255      7.692                                 
 5      97.87      33.333      4.255      7.547                                 
 6      87.81      37.500      6.383     10.909                                 
 7     341.72      25.000      2.128      3.922                                 
 8     169.64      40.000   

In [7]:
!prodigy ner.teach cord_19_abstracts_all_rf models/2020_03_30_all_RF/en_rf_sci_lg data/raw/cord_19_abstracts_filtered.jsonl \
  --label RISK_FACTOR

Using 1 label(s): RISK_FACTOR

✨  Starting the web server at http://localhost:8081 ...
Open the app in your browser and start annotating!

^C

✔ Saved 7 annotations to database SQLite
Dataset: cord_19_abstracts_all_rf
Session ID: 2020-03-30_22-01-26



# 2020.04.06 (Mon)

## Starting the annotations from scratch (`RF sentences`)

Following the official guide:
- https://www.youtube.com/watch?v=59BKHO_xBPA
- https://github.com/explosion/projects/tree/master/ner-food-ingredients#data-creation-and-training-workflow

### Label data manually

In [2]:
!prodigy ner.manual cord_19_rf_sentences blank:en data/raw/cord_19_rf_sentences.jsonl \
    --label RISK_FACTOR

Using 1 label(s): RISK_FACTOR

✨  Starting the web server at http://localhost:8080 ...
Open the app in your browser and start annotating!

^C

✔ Saved 102 annotations to database SQLite
Dataset: cord_19_rf_sentences
Session ID: 2020-04-06_22-08-56



### Pretrain different models

#### `en_vectors_web_lg`

In [5]:
!prodigy train ner cord_19_rf_sentences en_vectors_web_lg --eval-split 0.2

✔ Loaded model 'en_vectors_web_lg'
Created and merged data for 99 total examples
Using 80 train / 19 eval (split 20%)
Component: ner | Batch size: compounding | Dropout: 0.2 | Iterations: 10
ℹ Baseline accuracy: 0.000

=========================== ✨  Training the model ===========================

#    Loss       Precision   Recall     F-Score 
--   --------   ---------   --------   --------
 1     389.18       0.000      0.000      0.000                                 
 2     191.03      25.000      4.000      6.897                                 
 3     265.45      35.294     24.000     28.571                                 
 4     246.57      28.571     24.000     26.087                                 
 5     239.21      22.727     20.000     21.277                                 
 6     215.80      42.857     24.000     30.769                                 
 7     258.40      42.105     32.000     36.364                                 
 8     237.19      50.000     32.000   

#### `en_core_web_lg`

In [6]:
!prodigy train ner cord_19_rf_sentences en_core_web_lg --eval-split 0.2

✔ Loaded model 'en_core_web_lg'
Created and merged data for 99 total examples
Using 80 train / 19 eval (split 20%)
Component: ner | Batch size: compounding | Dropout: 0.2 | Iterations: 10
ℹ Baseline accuracy: 0.000

=========================== ✨  Training the model ===========================

#    Loss       Precision   Recall     F-Score 
--   --------   ---------   --------   --------
 1    1421.81       0.000      0.000      0.000                                 
 2    1245.16       0.000      0.000      0.000                                 
 3    1262.77       0.000      0.000      0.000                                 
 4    1249.04       0.000      0.000      0.000                                 
 5    1262.46       0.000      0.000      0.000                                 
 6    1205.26       0.000      0.000      0.000                                 
 7    1137.92       0.000      0.000      0.000                                 
 8    1131.84       0.000      0.000      

⁉️ Why is `en_core_web_lg` producing no results at all?

#### `en_core_sci_lg`

In [14]:
!prodigy train ner cord_19_rf_sentences en_core_sci_lg --eval-split 0.2 --output models/2020_04_06_rf_sentences

✔ Loaded model 'en_core_sci_lg'
Created and merged data for 99 total examples
Using 80 train / 19 eval (split 20%)
Component: ner | Batch size: compounding | Dropout: 0.2 | Iterations: 10
ℹ Baseline accuracy: 0.000

=========================== ✨  Training the model ===========================

#    Loss       Precision   Recall     F-Score 
--   --------   ---------   --------   --------
 1     559.17       0.000      0.000      0.000                                 
 2     550.33      33.333      8.000     12.903                                 
 3     538.43      50.000     24.000     32.432                                 
 4     457.89      52.632     40.000     45.455                                 
 5     452.06      50.000     36.000     41.860                                 
 6     415.01      60.000     48.000     53.333                                 
 7     489.45      47.826     44.000     45.833                                 
 8     461.13      45.000     36.000     4

#### `en_vectors_web_lg` + tok2vec

In [8]:
!prodigy train ner cord_19_rf_sentences en_vectors_web_lg --eval-split 0.2 \
    --init-tok2vec models/tok2vec_rf_vectors_model436.bin

✔ Loaded model 'en_vectors_web_lg'
Created and merged data for 99 total examples
Using 80 train / 19 eval (split 20%)
Component: ner | Batch size: compounding | Dropout: 0.2 | Iterations: 10
✔ Initializing with tok2vec weights
models/tok2vec_rf_vectors_model436.bin
embed_rows: 2000 | require_vectors: True | cnn_maxout_pieces: 3 |
token_vector_width: 96 | conv_depth: 4 | nr_feature_tokens: 3 |
pretrained_vectors: en_vectors_web_lg.vectors
ℹ Baseline accuracy: 0.000

=========================== ✨  Training the model ===========================

#    Loss       Precision   Recall     F-Score 
--   --------   ---------   --------   --------
 1     339.88       0.000      0.000      0.000                                 
 2     199.85      50.000     16.000     24.242                                 
 3     128.26      50.000     28.000     35.897                                 
 4      76.43      42.857     24.000     30.769                                 
 5      63.28      60.000     3

#### `en_core_web_lg` + tok2vec

In [9]:
!prodigy train ner cord_19_rf_sentences en_core_web_lg --eval-split 0.2 \
    --init-tok2vec models/tok2vec_rf_vectors_model436.bin

✔ Loaded model 'en_core_web_lg'
Created and merged data for 99 total examples
Using 80 train / 19 eval (split 20%)
Component: ner | Batch size: compounding | Dropout: 0.2 | Iterations: 10
✔ Initializing with tok2vec weights
models/tok2vec_rf_vectors_model436.bin

ℹ Baseline accuracy: 0.000

=========================== ✨  Training the model ===========================

#    Loss       Precision   Recall     F-Score 
--   --------   ---------   --------   --------
 1    2055.18       0.000      0.000      0.000                                 
 2    1808.48       0.000      0.000      0.000                                 
 3    1624.76       0.000      0.000      0.000                                 
 4    1443.31       0.000      0.000      0.000                                 
 5    1652.93       0.000      0.000      0.000                                 
 6    1470.22       0.000      0.000      0.000                                 
 7    1345.14       0.000      0.000      0.000

#### `en_core_sci_lg` + tok2vec

In [13]:
!prodigy train ner cord_19_rf_sentences en_core_sci_lg --eval-split 0.2 \
    --init-tok2vec models/tok2vec_rf_sci_model328.bin

✔ Loaded model 'en_core_sci_lg'
Created and merged data for 99 total examples
Using 80 train / 19 eval (split 20%)
Component: ner | Batch size: compounding | Dropout: 0.2 | Iterations: 10
✔ Initializing with tok2vec weights
models/tok2vec_rf_sci_model328.bin

ℹ Baseline accuracy: 0.000

=========================== ✨  Training the model ===========================

#    Loss       Precision   Recall     F-Score 
--   --------   ---------   --------   --------
 1     536.68       0.000      0.000      0.000                                 
 2     631.61       0.000      0.000      0.000                                 
 3     539.99      40.000     24.000     30.000                                 
 4     505.27      47.619     40.000     43.478                                 
 5     508.05      45.000     36.000     40.000                                 
 6     461.69      52.941     36.000     42.857                                 
 7     473.25      45.000     36.000     40.000    

### Label more data by correcting the model's predictions

The best model so far will be used: `Best F-Score   53.333`

In [15]:
!prodigy ner.correct cord_19_rf_sentences_correct models/2020_04_06_rf_sentences data/raw/cord_19_rf_sentences.jsonl \
    --label RISK_FACTOR --exclude cord_19_rf_sentences

Using 1 label(s): RISK_FACTOR
Added dataset cord_19_rf_sentences_correct to database SQLite.

✨  Starting the web server at http://localhost:8080 ...
Open the app in your browser and start annotating!

^C

✔ Saved 220 annotations to database SQLite
Dataset: cord_19_rf_sentences_correct
Session ID: 2020-04-06_23-58-48



### Train the final model

For futher training, only `en_core_sci_lg` will be used.

In [3]:
!prodigy train ner cord_19_rf_sentences,cord_19_rf_sentences_correct en_core_sci_lg --eval-split 0.2 --n-iter 20

✔ Loaded model 'en_core_sci_lg'
Created and merged data for 308 total examples
Using 247 train / 61 eval (split 20%)
Component: ner | Batch size: compounding | Dropout: 0.2 | Iterations: 20
ℹ Baseline accuracy: 0.000

=========================== ✨  Training the model ===========================

#    Loss       Precision   Recall     F-Score 
--   --------   ---------   --------   --------
 1    1554.25      61.538     37.500     46.602                                 
 2    1346.09      64.407     59.375     61.789                                 
 3    1407.15      66.667     50.000     57.143                                 
 4    1411.73      72.727     50.000     59.259                                 
 5    1412.81      73.913     53.125     61.818                                 
 6    1319.96      73.913     53.125     61.818                                 
 7    1378.05      72.917     54.688     62.500                                 
 8    1208.99      72.727     62.500    

In [6]:
!prodigy train ner cord_19_rf_sentences,cord_19_rf_sentences_correct en_core_sci_lg --eval-split 0.2 --n-iter 20 \
    --init-tok2vec models/tok2vec_rf_sci_model328.bin --output models/2020_04_07_rf_sentences_corrected

✔ Loaded model 'en_core_sci_lg'
Created and merged data for 308 total examples
Using 247 train / 61 eval (split 20%)
Component: ner | Batch size: compounding | Dropout: 0.2 | Iterations: 20
✔ Initializing with tok2vec weights
models/tok2vec_rf_sci_model328.bin

ℹ Baseline accuracy: 0.000

=========================== ✨  Training the model ===========================

#    Loss       Precision   Recall     F-Score 
--   --------   ---------   --------   --------
 1    1578.16      64.000     25.000     35.955                                 
 2    1439.61      55.319     40.625     46.847                                 
 3    1528.20      72.222     60.938     66.102                                 
 4    1429.18      71.154     57.812     63.793                                 
 5    1414.46      69.388     53.125     60.177                                 
 6    1344.69      75.472     62.500     68.376                                 
 7    1472.62      70.909     60.938     65.546  

In [5]:
!prodigy train ner cord_19_rf_sentences,cord_19_rf_sentences_correct en_core_sci_lg --eval-split 0.2 --n-iter 20 \
    --init-tok2vec models/tok2vec_rf_sci_model989.bin

✔ Loaded model 'en_core_sci_lg'
Created and merged data for 308 total examples
Using 247 train / 61 eval (split 20%)
Component: ner | Batch size: compounding | Dropout: 0.2 | Iterations: 20
✔ Initializing with tok2vec weights
models/tok2vec_rf_sci_model989.bin

ℹ Baseline accuracy: 0.000

=========================== ✨  Training the model ===========================

#    Loss       Precision   Recall     F-Score 
--   --------   ---------   --------   --------
 1    1570.28      62.500     23.438     34.091                                 
 2    1397.40      59.091     40.625     48.148                                 
 3    1485.47      70.833     53.125     60.714                                 
 4    1445.56      68.293     43.750     53.333                                 
 5    1431.31      64.000     50.000     56.140                                 
 6    1339.87      70.833     53.125     60.714                                 
 7    1387.07      72.340     53.125     61.261  

# 2020.04.09 (Thu)
## Experiments with new `tok2vec` models

In [1]:
!prodigy train ner cord_19_rf_sentences,cord_19_rf_sentences_correct en_core_sci_lg --eval-split 0.2 --n-iter 20 \
    --init-tok2vec models/tok2vec_rf_sci_model978_gpu.bin

✔ Loaded model 'en_core_sci_lg'
Created and merged data for 308 total examples
Using 247 train / 61 eval (split 20%)
Component: ner | Batch size: compounding | Dropout: 0.2 | Iterations: 20
✔ Initializing with tok2vec weights
models/tok2vec_rf_sci_model978_gpu.bin

ℹ Baseline accuracy: 0.000

=========================== ✨  Training the model ===========================

#    Loss       Precision   Recall     F-Score 
--   --------   ---------   --------   --------
 1    1577.68      57.895     34.375     43.137                                 
 2    1386.95      67.308     54.688     60.345                                 
 3    1455.49      76.596     56.250     64.865                                 
 4    1457.04      79.592     60.938     69.027                                 
 5    1413.30      78.431     62.500     69.565                                 
 6    1351.85      76.923     62.500     68.966                                 
 7    1304.38      78.000     60.938     68.4

In [2]:
!prodigy train ner cord_19_rf_sentences,cord_19_rf_sentences_correct en_core_sci_lg --eval-split 0.2 --n-iter 20 \
    --init-tok2vec models/tok2vec_rf_sci_model997_gpu.bin

✔ Loaded model 'en_core_sci_lg'
Created and merged data for 308 total examples
Using 247 train / 61 eval (split 20%)
Component: ner | Batch size: compounding | Dropout: 0.2 | Iterations: 20
✔ Initializing with tok2vec weights
models/tok2vec_rf_sci_model997_gpu.bin

ℹ Baseline accuracy: 0.000

=========================== ✨  Training the model ===========================

#    Loss       Precision   Recall     F-Score 
--   --------   ---------   --------   --------
 1    1630.03      55.556     23.438     32.967                                 
 2    1442.41      55.556     46.875     50.847                                 
 3    1429.19      65.957     48.438     55.856                                 
 4    1467.20      65.455     56.250     60.504                                 
 5    1467.20      69.643     60.938     65.000                                 
 6    1376.49      67.241     60.938     63.934                                 
 7    1324.06      73.585     60.938     66.6

In [3]:
!prodigy train ner cord_19_rf_sentences,cord_19_rf_sentences_correct en_core_sci_lg --eval-split 0.2 --n-iter 20 \
    --init-tok2vec models/tok2vec_abs_fil_sci_model975_gpu.bin

✔ Loaded model 'en_core_sci_lg'
Created and merged data for 308 total examples
Using 247 train / 61 eval (split 20%)
Component: ner | Batch size: compounding | Dropout: 0.2 | Iterations: 20
✔ Initializing with tok2vec weights
models/tok2vec_abs_fil_sci_model975_gpu.bin

ℹ Baseline accuracy: 0.000

=========================== ✨  Training the model ===========================

#    Loss       Precision   Recall     F-Score 
--   --------   ---------   --------   --------
 1    1587.95      62.963     26.562     37.363                                 
 2    1500.37      60.784     48.438     53.913                                 
 3    1510.36      63.492     62.500     62.992                                 
 4    1519.20      60.000     56.250     58.065                                 
 5    1521.26      65.574     62.500     64.000                                 
 6    1495.73      64.912     57.812     61.157                                 
 7    1386.50      65.517     59.375    

# 2020.04.10 (Fri)
## Experiments with new `tok2vec` models (cont.)

The Kaggle notebook training models for the full set of abstracts (i.e. `cord_19_abstracts.jsonl`) crashed, so I have only partial results and no loss metrics for them.

In [9]:
import os

from pathlib import Path

# run training for a selected models, use every skip_models-th file
skip_models = 3
abs_models_dir = Path('models/tok2vec_abs_sci_ALL_gpu')

models = sorted(os.listdir(abs_models_dir))

for idx, model_file in enumerate(models):
    if idx % skip_models == 0:
        model_path = abs_models_dir / model_file
        !prodigy train ner cord_19_rf_sentences,cord_19_rf_sentences_correct en_core_sci_lg \
            --eval-split 0.2 --n-iter 20 --init-tok2vec $model_path

✔ Loaded model 'en_core_sci_lg'
Created and merged data for 308 total examples
Using 247 train / 61 eval (split 20%)
Component: ner | Batch size: compounding | Dropout: 0.2 | Iterations: 20
✔ Initializing with tok2vec weights
models/tok2vec_abs_sci_ALL_gpu/model100.bin

ℹ Baseline accuracy: 0.000

=========================== ✨  Training the model ===========================

#    Loss       Precision   Recall     F-Score 
--   --------   ---------   --------   --------
 1    1570.29      43.478     31.250     36.364                                 
 2    1421.36      59.574     43.750     50.450                                 
 3    1486.19      68.519     57.812     62.712                                 
 4    1477.14      70.000     54.688     61.404                                 
 5    1389.42      73.469     56.250     63.717                                 
 6    1377.87      70.000     54.688     61.404                                 
 7    1379.27      65.385     53.125    

 5    1438.40      69.767     46.875     56.075                                 
 6    1374.64      72.340     53.125     61.261                                 
 7    1349.92      70.833     53.125     60.714                                 
 8    1265.91      73.585     60.938     66.667                                 
 9    1237.50      71.698     59.375     64.957                                 
10    1189.43      69.811     57.812     63.248                                 
11    1123.23      68.519     57.812     62.712                                 
12    1216.75      68.966     62.500     65.574                                 
13    1190.32      68.966     62.500     65.574                                 
14    1141.33      72.222     60.938     66.102                                 
15    1148.12      72.727     62.500     67.227                                 
16    1092.32      70.909     60.938     65.546                                 
17    1053.18      74.545   

15    1132.84      70.690     64.062     67.213                                 
16    1097.80      70.909     60.938     65.546                                 
17    1067.76      69.091     59.375     63.866                                 
18    1100.65      69.643     60.938     65.000                                 
19    1028.38      72.222     60.938     66.102                                 
20    1105.80      72.222     60.938     66.102                                 

============================= ✨  Results summary =============================

Label         Precision   Recall   F-Score
-----------   ---------   ------   -------
RISK_FACTOR      70.492   67.188    68.800


Best F-Score   68.800
Baseline       0.000              

✔ Loaded model 'en_core_sci_lg'
Created and merged data for 308 total examples
Using 247 train / 61 eval (split 20%)
Component: ner | Batch size: compounding | Dropout: 0.2 | Iterations: 20
✔ Initializing with tok2vec weights
models/tok2vec_abs

✔ Initializing with tok2vec weights
models/tok2vec_abs_sci_ALL_gpu/model130.bin

ℹ Baseline accuracy: 0.000

=========================== ✨  Training the model ===========================

#    Loss       Precision   Recall     F-Score 
--   --------   ---------   --------   --------
 1    1576.00      59.459     34.375     43.564                                 
 2    1414.30      60.417     45.312     51.786                                 
 3    1398.84      69.231     56.250     62.069                                 
 4    1483.87      72.222     60.938     66.102                                 
 5    1485.96      72.093     48.438     57.944                                 
 6    1377.94      72.000     56.250     63.158                                 
 7    1322.02      68.000     53.125     59.649                                 
 8    1241.86      67.347     51.562     58.407                                 
 9    1221.92      70.909     60.938     65.546                     

 7    1335.99      68.627     54.688     60.870                                 
 8    1413.31      64.706     51.562     57.391                                 
 9    1186.98      66.667     56.250     61.017                                 
10    1248.32      69.231     56.250     62.069                                 
11    1140.08      69.388     53.125     60.177                                 
12    1236.78      68.750     51.562     58.929                                 
13    1175.07      65.385     53.125     58.621                                 
14    1142.11      65.385     53.125     58.621                                 
15    1129.54      66.038     54.688     59.829                                 
16    1106.93      68.750     51.562     58.929                                 
17    1070.87      70.213     51.562     59.459                                 
18    1088.39      70.213     51.562     59.459                                 
19    1034.52      70.213   

17    1015.86      73.684     65.625     69.421                                 
18    1056.15      73.684     65.625     69.421                                 
19    1042.82      73.684     65.625     69.421                                 
20    1068.21      76.364     65.625     70.588                                 

============================= ✨  Results summary =============================

Label         Precision   Recall   F-Score
-----------   ---------   ------   -------
RISK_FACTOR      76.364   65.625    70.588


Best F-Score   70.588
Baseline       0.000              

✔ Loaded model 'en_core_sci_lg'
Created and merged data for 308 total examples
Using 247 train / 61 eval (split 20%)
Component: ner | Batch size: compounding | Dropout: 0.2 | Iterations: 20
✔ Initializing with tok2vec weights
models/tok2vec_abs_sci_ALL_gpu/model151.bin

ℹ Baseline accuracy: 0.000

=========================== ✨  Training the model ===========================

#    Loss       Precision   

ℹ Baseline accuracy: 0.000

=========================== ✨  Training the model ===========================

#    Loss       Precision   Recall     F-Score 
--   --------   ---------   --------   --------
 1    1576.79      57.143     37.500     45.283                                 
 2    1367.72      59.184     45.312     51.327                                 
 3    1447.57      66.667     59.375     62.810                                 
 4    1401.18      64.286     56.250     60.000                                 
 5    1415.15      66.667     50.000     57.143                                 
 6    1336.37      71.930     64.062     67.769                                 
 7    1349.04      71.154     57.812     63.793                                 
 8    1181.99      67.857     59.375     63.333                                 
 9    1174.85      70.909     60.938     65.546                                 
10    1216.98      72.727     62.500     67.227                     

 9    1226.29      68.966     62.500     65.574                                 
10    1216.42      69.492     64.062     66.667                                 
11    1164.83      71.667     67.188     69.355                                 
12    1277.11      71.667     67.188     69.355                                 
13    1209.09      73.333     68.750     70.968                                 
14    1150.05      72.414     65.625     68.852                                 
15    1110.54      71.667     67.188     69.355                                 
16    1104.07      70.492     67.188     68.800                                 
17    1057.22      68.852     65.625     67.200                                 
18    1087.16      68.254     67.188     67.717                                 
19    1036.89      68.254     67.188     67.717                                 
20    1091.13      69.355     67.188     68.254                                 

===========================

19    1029.31      65.517     59.375     62.295                                 
20    1091.05      67.797     62.500     65.041                                 

============================= ✨  Results summary =============================

Label         Precision   Recall   F-Score
-----------   ---------   ------   -------
RISK_FACTOR      76.364   65.625    70.588


Best F-Score   70.588
Baseline       0.000              

✔ Loaded model 'en_core_sci_lg'
Created and merged data for 308 total examples
Using 247 train / 61 eval (split 20%)
Component: ner | Batch size: compounding | Dropout: 0.2 | Iterations: 20
✔ Initializing with tok2vec weights
models/tok2vec_abs_sci_ALL_gpu/model181.bin

ℹ Baseline accuracy: 0.000

=========================== ✨  Training the model ===========================

#    Loss       Precision   Recall     F-Score 
--   --------   ---------   --------   --------
 1    1512.77      66.667     40.625     50.485                                 
 2    1377.63

 1    1508.05      64.516     31.250     42.105                                 
 2    1399.78      63.636     54.688     58.824                                 
 3    1447.90      62.069     56.250     59.016                                 
 4    1409.38      62.903     60.938     61.905                                 
 5    1412.41      69.492     64.062     66.667                                 
 6    1388.97      66.667     68.750     67.692                                 
 7    1319.59      62.500     54.688     58.333                                 
 8    1280.30      60.714     53.125     56.667                                 
 9    1215.82      60.714     53.125     56.667                                 
10    1188.50      60.714     53.125     56.667                                 
11    1171.68      67.742     65.625     66.667                                 
12    1224.73      70.690     64.062     67.213                                 
13    1199.54      70.690   

11    1158.58      71.698     59.375     64.957                                 
12    1256.12      73.077     59.375     65.517                                 
13    1179.84      72.000     56.250     63.158                                 
14    1109.52      71.429     54.688     61.947                                 
15    1107.24      74.074     62.500     67.797                                 
16    1129.53      73.077     59.375     65.517                                 
17    1025.24      71.154     57.812     63.793                                 
18    1097.39      70.588     56.250     62.609                                 
19    1053.17      73.077     59.375     65.517                                 
20    1118.76      71.698     59.375     64.957                                 

============================= ✨  Results summary =============================

Label         Precision   Recall   F-Score
-----------   ---------   ------   -------
RISK_FACTOR      74.545

✔ Loaded model 'en_core_sci_lg'
Created and merged data for 308 total examples
Using 247 train / 61 eval (split 20%)
Component: ner | Batch size: compounding | Dropout: 0.2 | Iterations: 20
✔ Initializing with tok2vec weights
models/tok2vec_abs_sci_ALL_gpu/model211.bin

ℹ Baseline accuracy: 0.000

=========================== ✨  Training the model ===========================

#    Loss       Precision   Recall     F-Score 
--   --------   ---------   --------   --------
 1    1577.06      47.222     26.562     34.000                                 
 2    1402.29      71.154     57.812     63.793                                 
 3    1436.70      69.091     59.375     63.866                                 
 4    1574.95      68.750     51.562     58.929                                 
 5    1455.37      67.308     54.688     60.345                                 
 6    1369.23      66.038     54.688     59.829                                 
 7    1319.05      59.322     54.688    

 5    1369.22      73.585     60.938     66.667                                 
 6    1395.24      71.429     54.688     61.947                                 
 7    1343.63      66.667     53.125     59.130                                 
 8    1254.10      63.158     56.250     59.504                                 
 9    1255.24      64.815     54.688     59.322                                 
10    1169.19      74.576     68.750     71.545                                 
11    1158.29      72.414     65.625     68.852                                 
12    1220.35      72.414     65.625     68.852                                 
13    1174.16      74.138     67.188     70.492                                 
14    1121.06      74.545     64.062     68.908                                 
15    1088.34      71.186     65.625     68.293                                 
16    1129.11      70.690     64.062     67.213                                 
17    1070.65      72.727   

15    1134.67      70.909     60.938     65.546                                 
16    1104.98      69.091     59.375     63.866                                 
17    1036.33      67.273     57.812     62.185                                 
18    1072.51      67.857     59.375     63.333                                 
19    1039.27      71.698     59.375     64.957                                 
20    1074.77      71.698     59.375     64.957                                 

============================= ✨  Results summary =============================

Label         Precision   Recall   F-Score
-----------   ---------   ------   -------
RISK_FACTOR      75.472   62.500    68.376


Best F-Score   68.376
Baseline       0.000              

✔ Loaded model 'en_core_sci_lg'
Created and merged data for 308 total examples
Using 247 train / 61 eval (split 20%)
Component: ner | Batch size: compounding | Dropout: 0.2 | Iterations: 20
✔ Initializing with tok2vec weights
models/tok2vec_abs

✔ Initializing with tok2vec weights
models/tok2vec_abs_sci_ALL_gpu/model241.bin

ℹ Baseline accuracy: 0.000

=========================== ✨  Training the model ===========================

#    Loss       Precision   Recall     F-Score 
--   --------   ---------   --------   --------
 1    1519.45      51.429     28.125     36.364                                 
 2    1436.86      50.909     43.750     47.059                                 
 3    1510.82      67.797     62.500     65.041                                 
 4    1392.90      64.912     57.812     61.157                                 
 5    1426.40      62.264     51.562     56.410                                 
 6    1431.06      66.667     65.625     66.142                                 
 7    1377.47      66.667     65.625     66.142                                 
 8    1238.80      67.188     67.188     67.188                                 
 9    1215.15      66.667     65.625     66.142                     

 7    1328.60      76.471     60.938     67.826                                 
 8    1220.65      76.596     56.250     64.865                                 
 9    1235.62      72.340     53.125     61.261                                 
10    1262.71      72.340     53.125     61.261                                 
11    1204.17      72.000     56.250     63.158                                 
12    1251.53      72.000     56.250     63.158                                 
13    1155.18      72.917     54.688     62.500                                 
14    1109.41      72.917     54.688     62.500                                 
15    1119.35      74.468     54.688     63.063                                 
16    1106.50      73.469     56.250     63.717                                 
17    1030.60      74.468     54.688     63.063                                 
18    1053.18      72.917     54.688     62.500                                 
19    1041.28      72.917   

17    1073.30      73.585     60.938     66.667                                 
18    1067.49      73.585     60.938     66.667                                 
19    1059.54      75.000     60.938     67.241                                 
20    1117.32      73.585     60.938     66.667                                 

============================= ✨  Results summary =============================

Label         Precision   Recall   F-Score
-----------   ---------   ------   -------
RISK_FACTOR      77.193   68.750    72.727


Best F-Score   72.727
Baseline       0.000              

✔ Loaded model 'en_core_sci_lg'
Created and merged data for 308 total examples
Using 247 train / 61 eval (split 20%)
Component: ner | Batch size: compounding | Dropout: 0.2 | Iterations: 20
✔ Initializing with tok2vec weights
models/tok2vec_abs_sci_ALL_gpu/model262.bin

ℹ Baseline accuracy: 0.000

=========================== ✨  Training the model ===========================

#    Loss       Precision   

ℹ Baseline accuracy: 0.000

=========================== ✨  Training the model ===========================

#    Loss       Precision   Recall     F-Score 
--   --------   ---------   --------   --------
 1    1495.66      75.862     34.375     47.312                                 
 2    1510.79      66.667     56.250     61.017                                 
 3    1552.58      69.643     60.938     65.000                                 
 4    1410.60      66.667     53.125     59.130                                 
 5    1405.83      65.217     46.875     54.545                                 
 6    1357.83      64.583     48.438     55.357                                 
 7    1320.01      64.000     50.000     56.140                                 
 8    1221.06      63.462     51.562     56.897                                 
 9    1216.96      62.000     48.438     54.386                                 
10    1174.03      69.643     60.938     65.000                     

 9    1217.85      71.186     65.625     68.293                                 
10    1185.69      70.690     64.062     67.213                                 
11    1114.42      68.333     64.062     66.129                                 
12    1234.75      70.690     64.062     67.213                                 
13    1175.35      73.684     65.625     69.421                                 
14    1138.86      73.214     64.062     68.333                                 
15    1078.85      73.684     65.625     69.421                                 
16    1084.59      71.429     62.500     66.667                                 
17    1007.19      71.429     62.500     66.667                                 
18    1059.53      71.930     64.062     67.769                                 
19    1024.42      73.684     65.625     69.421                                 
20    1090.38      71.930     64.062     67.769                                 

===========================

19    1053.35      72.222     60.938     66.102                                 
20    1068.43      71.154     57.812     63.793                                 

============================= ✨  Results summary =============================

Label         Precision   Recall   F-Score
-----------   ---------   ------   -------
RISK_FACTOR      71.667   67.188    69.355


Best F-Score   69.355
Baseline       0.000              

✔ Loaded model 'en_core_sci_lg'
Created and merged data for 308 total examples
Using 247 train / 61 eval (split 20%)
Component: ner | Batch size: compounding | Dropout: 0.2 | Iterations: 20
✔ Initializing with tok2vec weights
models/tok2vec_abs_sci_ALL_gpu/model292.bin

ℹ Baseline accuracy: 0.000

=========================== ✨  Training the model ===========================

#    Loss       Precision   Recall     F-Score 
--   --------   ---------   --------   --------
 1    1506.33      76.667     35.938     48.936                                 
 2    1413.90

 1    1516.87      69.231     28.125     40.000                                 
 2    1442.83      61.111     51.562     55.932                                 
 3    1413.95      71.930     64.062     67.769                                 
 4    1403.35      75.000     70.312     72.581                                 
 5    1385.03      75.862     68.750     72.131                                 
 6    1296.83      74.576     68.750     71.545                                 
 7    1358.15      75.862     68.750     72.131                                 
 8    1220.61      70.968     68.750     69.841                                 
 9    1190.69      73.333     68.750     70.968                                 
10    1229.27      73.684     65.625     69.421                                 
11    1157.94      78.846     64.062     70.690                                 
12    1220.14      74.545     64.062     68.908                                 
13    1139.34      73.214   

11    1110.71      70.492     67.188     68.800                                 
12    1193.98      72.881     67.188     69.919                                 
13    1163.38      66.667     65.625     66.142                                 
14    1124.38      65.152     67.188     66.154                                 
15    1103.28      67.742     65.625     66.667                                 
16    1076.95      71.186     65.625     68.293                                 
17    1064.07      72.414     65.625     68.852                                 
18    1070.69      72.414     65.625     68.852                                 
19    1037.43      67.213     64.062     65.600                                 
20    1103.58      67.213     64.062     65.600                                 

============================= ✨  Results summary =============================

Label         Precision   Recall   F-Score
-----------   ---------   ------   -------
RISK_FACTOR      74.576

✔ Loaded model 'en_core_sci_lg'
Created and merged data for 308 total examples
Using 247 train / 61 eval (split 20%)
Component: ner | Batch size: compounding | Dropout: 0.2 | Iterations: 20
✔ Initializing with tok2vec weights
models/tok2vec_abs_sci_ALL_gpu/model322.bin

ℹ Baseline accuracy: 0.000

=========================== ✨  Training the model ===========================

#    Loss       Precision   Recall     F-Score 
--   --------   ---------   --------   --------
 1    1515.87      53.659     34.375     41.905                                 
 2    1491.23      50.000     43.750     46.667                                 
 3    1444.48      58.182     50.000     53.782                                 
 4    1493.63      63.793     57.812     60.656                                 
 5    1441.77      65.000     60.938     62.903                                 
 6    1337.14      62.903     60.938     61.905                                 
 7    1281.15      66.667     65.625    

 5    1382.70      71.667     67.188     69.355                                 
 6    1334.54      73.684     65.625     69.421                                 
 7    1310.78      74.138     67.188     70.492                                 
 8    1225.78      72.881     67.188     69.919                                 
 9    1178.12      74.074     62.500     67.797                                 
10    1150.85      75.472     62.500     68.376                                 
11    1130.27      75.926     64.062     69.492                                 
12    1221.05      77.778     65.625     71.186                                 
13    1173.35      77.778     65.625     71.186                                 
14    1117.11      76.364     65.625     70.588                                 
15    1087.97      76.364     65.625     70.588                                 
16    1102.23      75.000     65.625     70.000                                 
17    1041.93      72.881   

15    1081.60      71.930     64.062     67.769                                 
16    1082.05      71.930     64.062     67.769                                 
17    1057.24      71.930     64.062     67.769                                 
18    1055.70      73.684     65.625     69.421                                 
19    1034.63      77.358     64.062     70.085                                 
20    1095.07      77.358     64.062     70.085                                 

============================= ✨  Results summary =============================

Label         Precision   Recall   F-Score
-----------   ---------   ------   -------
RISK_FACTOR      77.358   64.062    70.085


Best F-Score   70.085
Baseline       0.000              

✔ Loaded model 'en_core_sci_lg'
Created and merged data for 308 total examples
Using 247 train / 61 eval (split 20%)
Component: ner | Batch size: compounding | Dropout: 0.2 | Iterations: 20
✔ Initializing with tok2vec weights
models/tok2vec_abs

✔ Initializing with tok2vec weights
models/tok2vec_abs_sci_ALL_gpu/model352.bin

ℹ Baseline accuracy: 0.000

=========================== ✨  Training the model ===========================

#    Loss       Precision   Recall     F-Score 
--   --------   ---------   --------   --------
 1    1501.57      67.742     32.812     44.211                                 
 2    1396.89      56.452     54.688     55.556                                 
 3    1375.31      68.966     62.500     65.574                                 
 4    1390.81      75.000     65.625     70.000                                 
 5    1321.72      73.214     64.062     68.333                                 
 6    1338.10      71.429     62.500     66.667                                 
 7    1332.88      70.000     65.625     67.742                                 
 8    1290.72      71.186     65.625     68.293                                 
 9    1204.20      67.213     64.062     65.600                     

 7    1354.79      69.492     64.062     66.667                                 
 8    1216.38      70.000     65.625     67.742                                 
 9    1191.40      71.930     64.062     67.769                                 
10    1213.28      69.492     64.062     66.667                                 
11    1120.01      72.414     65.625     68.852                                 
12    1184.18      70.690     64.062     67.213                                 
13    1168.83      70.000     65.625     67.742                                 
14    1137.41      71.186     65.625     68.293                                 
15    1096.33      72.881     67.188     69.919                                 
16    1110.22      72.881     67.188     69.919                                 
17    1042.73      70.000     65.625     67.742                                 
18    1085.84      70.000     65.625     67.742                                 
19    1051.89      71.186   

17    1061.78      71.186     65.625     68.293                                 
18    1051.42      65.079     64.062     64.567                                 
19    1030.53      62.500     62.500     62.500                                 
20    1079.54      62.500     62.500     62.500                                 

============================= ✨  Results summary =============================

Label         Precision   Recall   F-Score
-----------   ---------   ------   -------
RISK_FACTOR      71.186   65.625    68.293


Best F-Score   68.293
Baseline       0.000              

✔ Loaded model 'en_core_sci_lg'
Created and merged data for 308 total examples
Using 247 train / 61 eval (split 20%)
Component: ner | Batch size: compounding | Dropout: 0.2 | Iterations: 20
✔ Initializing with tok2vec weights
models/tok2vec_abs_sci_ALL_gpu/model373.bin

ℹ Baseline accuracy: 0.000

=========================== ✨  Training the model ===========================

#    Loss       Precision   

ℹ Baseline accuracy: 0.000

=========================== ✨  Training the model ===========================

#    Loss       Precision   Recall     F-Score 
--   --------   ---------   --------   --------
 1    1452.25      76.923     31.250     44.444                                 
 2    1457.58      58.182     50.000     53.782                                 
 3    1478.05      55.738     53.125     54.400                                 
 4    1438.49      69.492     64.062     66.667                                 
 5    1347.63      71.667     67.188     69.355                                 
 6    1348.81      70.690     64.062     67.213                                 
 7    1324.75      70.690     64.062     67.213                                 
 8    1294.85      70.690     64.062     67.213                                 
 9    1210.03      70.909     60.938     65.546                                 
10    1158.22      70.909     60.938     65.546                     

 9    1182.15      71.930     64.062     67.769                                 
10    1244.71      72.881     67.188     69.919                                 
11    1107.36      70.492     67.188     68.800                                 
12    1196.49      72.131     68.750     70.400                                 
13    1168.93      75.000     70.312     72.581                                 
14    1169.17      76.271     70.312     73.171                                 
15    1097.94      75.000     70.312     72.581                                 
16    1106.83      75.000     70.312     72.581                                 
17    1035.00      76.667     71.875     74.194                                 
18    1086.49      75.000     70.312     72.581                                 
19    1051.42      76.271     70.312     73.171                                 
20    1094.91      71.667     67.188     69.355                                 

===========================

19    1049.51      70.000     54.688     61.404                                 
20    1108.60      67.925     56.250     61.538                                 

============================= ✨  Results summary =============================

Label         Precision   Recall   F-Score
-----------   ---------   ------   -------
RISK_FACTOR      81.132   67.188    73.504


Best F-Score   73.504
Baseline       0.000              

✔ Loaded model 'en_core_sci_lg'
Created and merged data for 308 total examples
Using 247 train / 61 eval (split 20%)
Component: ner | Batch size: compounding | Dropout: 0.2 | Iterations: 20
✔ Initializing with tok2vec weights
models/tok2vec_abs_sci_ALL_gpu/model402.bin

ℹ Baseline accuracy: 0.000

=========================== ✨  Training the model ===========================

#    Loss       Precision   Recall     F-Score 
--   --------   ---------   --------   --------
 1    1488.26      58.140     39.062     46.729                                 
 2    1421.99